independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/strategy/SmartPomMergeCodestartFileStrategyHandler.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.devtools.codestarts.strategy;
+
+import io.fabric8.maven.Maven;
+import io.fabric8.maven.merge.SmartModelMerger;
+import io.quarkus.devtools.codestarts.reader.CodestartFile;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import org.apache.maven.model.Model;
+
+final class SmartPomMergeCodestartFileStrategyHandler implements CodestartFileStrategyHandler {
```

Comment:
do we have a usecase for this strategy ? I'm sure there is one i just can't figure one out atm ;)

Reply:
Yes, thanks to this we could have a Red Hat `tooling` codestart adding `<repositories>` for example.